CONTRIBUTING.md

```diff
@@ -92,12 +106,121 @@ Commit messages are linted on commit, so you'll know if your message is not quit
 
 ## Setting up the dev environment
 
-After cloning the Vendure repo, please follow the [Development guide](https://github.com/vendure-ecommerce/vendure/blob/master/README.md#development) in the README for instructions on how to get up and running locally.
+After [cloning the Vendure repo](#cloning-locally), please refer to the [Development guide](https://github.com/vendure-ecommerce/vendure/blob/master/README.md#development) in the README for instructions on setting up your local development environment.
+
+## Suggested contribution workflow
+
+### Creating a fork:
+
+Start by creating a fork of the repository. This can be done by navigating to our [repository](https://github.com/vendure-ecommerce/vendure) and clicking on the fork button. This creates your own copy under your GitHub account. We recommend keeping "vendure" as the name of the fork.
+
+### Cloning locally: 
+Run the following commands to create a local clone of the repository files, where you can safely develop and test out your changes:
+
+```bash
+git clone https://github.com/YOUR-USERNAME/vendure.git
+cd vendure
+```
+
+Add the [Vendure repository](https://github.com/vendure-ecommerce/vendure) as the upstream remote to your clone:
+
+```bash
+git remote add upstream https://github.com/vendure-ecommerce/vendure.git
+```
+
+This lets you pull updates from the original repository directly to your local clone.
+
+### Staying up to date
+> [!TIP]
+> It is a good idea to regularly update your local and forked repositories. You can do so by running the following commands:
+> ```bash
+> git fetch upstream
+> git checkout master
+> ```
+> Now you're on the master branch of your local repository, and Git "knows" whether there are differences between your master branch and the upstream branch.
+>
+> To update your local and forked remote repositories: 
+> ```bash
+> git merge upstream/master
+> git push origin master
+> ```
+
+### Create a new branch
+
+This is where you can start implementing your changes:
+
+```bash
+git checkout -b your-new-branch
+```
+
+### Developing
+
+Follow our [development guide](https://github.com/vendure-ecommerce/vendure#development) to make sure you have a properly set up development environment.
+
+After implementing your changes, stage the changes and commit them. Refer to the [Commit message format](#commit-message-format)
+```
+git add .
+git commit -m "type(scope): Message in present tense"
+```
+
+### Creating a pull request
+
+> [!TIP]
+> Now would be a good time to [update](#staying-up-to-date) your local and remote repositories.
 
-## Contributing Admin UI translations
+Make sure you have your new branch checked out and rebase it onto your up-to-date master branch:
+```bash
+git checkout your-new-branch
```

Comment:
Do we want to recommend rebase rather than merge? I personally find rebase can sometimes be way more problematic when you need to re-resolve conflicts of several subsequent commits. With a merge it's always just a single conflict resolution.

Since we will squash and merge the PR, the end result will always be a single commit anyway.